lib/rig_inbound_gateway_web/v1/metadata_controller.ex

```diff
@@ -0,0 +1,178 @@
+        locale
+        timezone
+
+  All indexed fields are then being propagated to all other instances, so they know where to find the corresponding metadata.
+
+  ## Example
+
+  A user connects to RIG 1 and sends the metadata fields from above to RIG 1. RIG 1 would propagate the connection token and the userid along with the RIG instance ID to all other RIG instances.
+  If someone were to access the metadata of a user from RIG 3, RIG 3 would know where to find the metadata, ask RIG 1 for it and return the metadata set.
+
+  ## Dirty testing with JWT
+
+      AUTH='Authorization:Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiI5YWIxYmZmMi1hOGQ4LTQ1NWMtYjQ4YS01MDE0NWQ3ZDhlMzAiLCJuYW1lIjoiSm9obiBEb2UiLCJpYXQiOjE1Njg3MTMzNjEsImV4cCI6NDEwMzI1ODE0M30.kjiR7kFyOEeMJaY1zPCctut39eEWmKswUCNZdK5Q3-w'
+      META='{ "metadata": { "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META" "$AUTH"
+
+  ## Dirty testing without JWT
+      META='{ "metadata": { "userid": "9ab1bff2-a8d8-455c-b48a-50145d7d8e30", "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META"
+
+  """
+  @spec set_metadata(conn :: Plug.Conn.t(), params :: map) :: Plug.Conn.t()
+  def set_metadata(
+        %{method: "PUT"} = conn,
+        %{
+          "connection_id" => connection_id
+        }
+      ) do
+
+    conn
+    |> accept_only_req_for(["application/json"])
+    |> decode_metadata()
+    |> do_set_metadata(connection_id)
+  end
+
+  defp decode_metadata(conn) do
+    with {"application", "json"} <- content_type(conn),
+         {:ok, body, conn} <- BodyReader.read_full_body(conn),
+         {:ok, json} <- Jason.decode(body),
+         {:parse, %{"metadata" => metadata}} <- {:parse, json} do
+
+      conf = config()
+      jwt_fields = conf.jwt_fields
+
+      with {:ok, auth_tokens} <- Map.get(conn.assigns, :auth_tokens, [])
+      |> Enum.map(fn
+        {"bearer", token} -> JWT.parse_token(token)
+        _ -> {:ok, %{}}
+      end)
+      |> Result.list_to_result()
+      |> Result.map_err(fn decode_errors -> {:error, decode_errors} end) do
+
+        # Merge the list of maps into a single map
+        auth_tokens = Enum.reduce(auth_tokens, fn x, y ->
+          Map.merge(x, y, fn _k, v1, v2 -> v2 ++ v1 end)
+        end)
+
+        # Get values from JWT
+        jwt_vals = jwt_fields
+        |> Enum.map(fn x ->
+          key = x |> elem(0)
+          jwt_key = x |> elem(1)
+          {key, auth_tokens[jwt_key]}
+        end)
+        |> Enum.into(%{})
+
+        # Merge values from metadata with the values from JWT, prioritizing JWT values
+        metadata = Map.merge(metadata, jwt_vals)
+
+        conn
+        |> Plug.Conn.assign(:metadata, metadata)
+      else
+        {:error, _} ->
+          conn
+          |> Plug.Conn.assign(:metadata, metadata)
+      end
+    else
+      {:parse, json} ->
+        message = """
```

Comment:
The function is called `decode_metadata` and not `decode_metadata_then_assign_it_to_conn_but_if_there_is_an_error_emit_a_bad_request_response`. Consider separating the build-up of `metadata` (see comment above) and handling of `conn`.